### ISSUE TITLE
needless_doctest_main: false positive with proc-macro attribute

### ISSUE THREAD

**Issue by jonhoo** [rust-lang/rust-clippy#5188]
```console
$ cargo clippy -V
clippy 0.0.212 (ea85b4c 2020-02-01)
```

The `needless_doctest_main` lint triggers on this code:
```rust
/// ```
/// #[tokio::main]
/// async fn main() {
///     // ...
/// }
/// ```
```
Where the `fn main` is very much not needless :)

**Comment by llogiq** [comment]
Do you mean the given code in a doctest? Or just the code without any docs?

**Comment by jonhoo** [comment]
Ah, sorry, yes, the code in a doctest -- will update.

**Comment by llogiq** [comment]
Thank you! This is just a matter of adding `"async "` to the list of terms that the lint checks against.

**Comment by jonhoo** [comment]
@llogiq :+1: Pushed fix in https://github.com/rust-lang/rust-clippy/pull/5189.

**Comment by llogiq** [comment]
Thank you!
